deploy/examples/create-external-cluster-resources.py

```diff
@@ -1807,6 +1817,19 @@ def upgrade_user_permissions(self, user):
             print(f"user {user} not found for upgrading.")
             return
         existing_caps = json_out[0]["caps"]
+        if user == "client.healthchecker":
+            # when admin has not provided rgw pool name during upgrade,
+            # get the rgw pool name from client.healthchecker user which was used during connection
+            if not self._arg_parser.rgw_pool_prefix:
```

Comment:
can we separate this code in a function, the original function is becoming a lot big

Reply:
done